Sources/HTTPTypes/HTTPRequest.swift

```diff
@@ -146,53 +146,156 @@ public struct HTTPRequest: Sendable, Hashable {
-            willSet {
+            get {
+                self._storage.method
+            }
+            set {
                 precondition(newValue.name == .method, "Cannot change pseudo-header field name")
                 precondition(HTTPField.isValidToken(newValue.rawValue._storage), "Invalid character in method field")
+
+                if !isKnownUniquelyReferenced(&self._storage) {
+                    self._storage = self._storage.copy()
+                }
+                self._storage.method = newValue
             }
         }
 
         /// The underlying ":scheme" pseudo header field.
         public var scheme: HTTPField? {
-            willSet {
+            get {
+                self._storage.scheme
+            }
+            set {
                 if let name = newValue?.name {
                     precondition(name == .scheme, "Cannot change pseudo-header field name")
                 }
+
+                if !isKnownUniquelyReferenced(&self._storage) {
+                    self._storage = self._storage.copy()
+                }
+                self._storage.scheme = newValue
             }
         }
 
         /// The underlying ":authority" pseudo header field.
         public var authority: HTTPField? {
-            willSet {
+            get {
+                self._storage.authority
+            }
+            set {
                 if let name = newValue?.name {
                     precondition(name == .authority, "Cannot change pseudo-header field name")
                 }
+
+                if !isKnownUniquelyReferenced(&self._storage) {
+                    self._storage = self._storage.copy()
+                }
+                self._storage.authority = newValue
             }
         }
 
         /// The underlying ":path" pseudo header field.
         public var path: HTTPField? {
-            willSet {
+            get {
+                self._storage.path
+            }
+            set {
                 if let name = newValue?.name {
                     precondition(name == .path, "Cannot change pseudo-header field name")
                 }
+
+                if !isKnownUniquelyReferenced(&self._storage) {
+                    self._storage = self._storage.copy()
+                }
+                self._storage.path = newValue
             }
         }
 
         /// The underlying ":protocol" pseudo header field.
         public var extendedConnectProtocol: HTTPField? {
-            willSet {
+            get {
+                self._storage.extendedConnectProtocol
+            }
+            set {
                 if let name = newValue?.name {
                     precondition(name == .protocol, "Cannot change pseudo-header field name")
                 }
+
+                if !isKnownUniquelyReferenced(&self._storage) {
+                    self._storage = self._storage.copy()
+                }
+                self._storage.extendedConnectProtocol = newValue
```

Comment:
A bit of a nit, but these accessors probably want to either replace `set` with `_modify` or to use `@inlinable`. Otherwise minor modifications can risk CoWing the underlying strings where it isn't necessary to do so.

This isn't a _huge_ deal as I don't think there's a lot of minor string modifications happening this way (pretty rare to do `request.path.value += "/something/else"`) but it'd be nice to avoid the CoW in that case.

Reply:
I'm not very familiar with `_modify`. Does this look reasonable?

```
            _modify {
                if !isKnownUniquelyReferenced(&self._storage) {
                    self._storage = self._storage.copy()
                }
                yield &self._storage.path
                if let name = self._storage.path?.name {
                    precondition(name == .path, "Cannot change pseudo-header field name")
                }
            }
```

Reply:
I _believe_ that will be fine, though `_modify` is extremely sensitive to the exact nature of its flow control. I recommend actually building this under `-O` and checking that there isn't a call to `malloc` or similar there.

Reply:
There is a call to `malloc` after this change. Not sure how to get rid of it.

[libHTTPTypes.dylib.zip](https://github.com/user-attachments/files/18834010/libHTTPTypes.dylib.zip)

<img width="2039" alt="Screenshot 2025-02-17 at 3 40 44 PM" src="https://github.com/user-attachments/assets/b690a1a2-5cac-41cb-ac62-c9a13c889596" />

Reply:
Will take a look